sycl/include/sycl/accessor.hpp

```diff
@@ -1145,6 +1145,33 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
       : impl({}, detail::InitializedVal<AdjustedDim, range>::template get<0>(),
              detail::InitializedVal<AdjustedDim, range>::template get<0>()) {}
 
+  // implicit conversion between const / non-const types for read only accessors
+  template <typename DataT_, int Dimensions_, access::mode AccessMode_,
+            access::target AccessTarget_, access::placeholder IsPlaceholder_,
+            typename PropertyListT_>
+  accessor(const accessor<DataT_, Dimensions_, AccessMode_, AccessTarget_,
+                          IsPlaceholder_, PropertyListT_> &other,
+           std::enable_if_t<(AccessMode_ == access_mode::read) &&
+                            !(std::is_same<DataT_, DataT>::value) &&
+                            std::is_same<std::remove_const_t<DataT_>,
+                                         std::remove_const_t<DataT>>::value> * =
+               nullptr)
+      : impl(other.impl) {}
+
+  // implicit conversion from read_write T accessor to read only T (const)
+  // accessor
+  template <typename DataT_, int Dimensions_, access::mode AccessMode_,
+            access::target AccessTarget_, access::placeholder IsPlaceholder_,
+            typename PropertyListT_>
+  accessor(const accessor<DataT_, Dimensions_, AccessMode_, AccessTarget_,
+                          IsPlaceholder_, PropertyListT_> &other,
+           std::enable_if_t<(AccessMode_ == access_mode::read_write) &&
```

Comment:
```suggestion
  template <typename DataT_>
  accessor(const accessor<DataT_, Dimensions, access::mode::read_write, AccessTarget,
                          IsPlaceholder, PropertyListT> &other,
           std::enable_if_t<
```
This is in accordance with [Section 4.7.6.9.3.](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessor.command.buffer.conversions) stating that

> There is also an implicit conversion from the read-write specialization shown in [Table 60](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.accessors.command.buffer.read-write) to any of the read-only specializations shown in [Table 59](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.accessors.command.buffer.read-only), **provided that all other template parameters are the same.**

Reply:
thanks @steffenlarsen, you are right. Done.